Ruff has issues, and I'm happy to use this PR as guinea pig for it.

I can reproduce by running check_compliance locally:
```
tinyvision@vm1:~/zephyrproject/zephyr$ python scripts/ci/check_compliance.py ^G
Running BinaryFiles      tests in /home/tinyvision/zephyrproject/zephyr ...
Running Kconfig          tests in /home/tinyvision/zephyrproject/zephyr ...
Running YAMLLint         tests in /home/tinyvision/zephyrproject/zephyr ...
Running Identity         tests in /home/tinyvision/zephyrproject/zephyr ...
Running ModulesMaintainers tests in /home/tinyvision/zephyrproject/zephyr ...
Running Pylint           tests in /home/tinyvision/zephyrproject/zephyr ...
Running MaintainersFormat tests in /home/tinyvision/zephyrproject/zephyr ...
Running DevicetreeBindings tests in /home/tinyvision/zephyrproject/zephyr ...
Running Gitlint          tests in /home/tinyvision/zephyrproject/zephyr ...
Running BoardYml         tests in /home/tinyvision/zephyrproject/zephyr ...
Running KconfigBasicNoModules tests in /home/tinyvision/zephyrproject/zephyr ...
Running TextEncoding     tests in /home/tinyvision/zephyrproject/zephyr ...
Running GitDiffCheck     tests in /home/tinyvision/zephyrproject/zephyr ...
Running Ruff             tests in /home/tinyvision/zephyrproject/zephyr ...
warning: The following rules have been removed and ignoring them has no effect:
    - UP027

Python exception in `/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py`:

```
Traceback (most recent call last):
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1936, in main
    n_fails = _main(args)
              ^^^^^^^^^^^
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1871, in _main
    test.run()
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1666, in run
    messages = json.loads(output)
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)

```
Traceback (most recent call last):
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1963, in <module>
    main(sys.argv[1:])
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1936, in main
    n_fails = _main(args)
              ^^^^^^^^^^^
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1871, in _main
    test.run()
  File "/home/tinyvision/zephyrproject/zephyr/scripts/ci/check_compliance.py", line 1666, in run
    messages = json.loads(output)
               ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

```
(py) tinyvision@vm1:~/zephyrproject/zephyr$ uname -a
Linux vm1 6.8.0-49-generic #49-Ubuntu SMP PREEMPT_DYNAMIC Mon Nov  4 02:06:24 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```
```
(py) tinyvision@vm1:~/zephyrproject/zephyr$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04 LTS
Release:        24.04
Codename:       noble
```
```
(py) tinyvision@vm1:~/zephyrproject/zephyr$ python --version
Python 3.12.3
```

--- a/boards/common/ecpprog.board.cmake
+++ b/boards/common/ecpprog.board.cmake
@@ -1,0 +1,5 @@
+# Copyright (c) 2024 tinyVision.ai Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+board_set_flasher_ifnset(ecpprog)
+board_finalize_runner_args(ecpprog)

--- a/scripts/west_commands/runners/__init__.py
+++ b/scripts/west_commands/runners/__init__.py
@@ -31,6 +31,7 @@ _names = [
     'canopen_program',
     'dediprog',
     'dfu',
+    'ecpprog',
     'esp32',
     'ezflashcli',
     'gd32isp',

--- a/scripts/west_commands/runners/ecpprog.py
+++ b/scripts/west_commands/runners/ecpprog.py
@@ -1,0 +1,41 @@
+# Copyright (c) 2024 tinyVision.ai Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Runner for the ecpprog programming tool for Lattice FPGAs."""
+# https://github.com/gregdavill/ecpprog
+
+from runners.core import BuildConfiguration, RunnerCaps, ZephyrBinaryRunner
+
+
+class EcpprogBinaryRunner(ZephyrBinaryRunner):
+    """Runner front-end for programming the FPGA flash at some offset."""
+
+    def __init__(self, cfg, device=None):
+        super().__init__(cfg)
+        self.device = device
+
+    @classmethod
+    def name(cls):
+        return "ecpprog"
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={"flash"})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        parser.add_argument(
+            "--device", dest="device", help="Device identifier such as i:<vid>:<pid>"
+        )
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return EcpprogBinaryRunner(cfg, device=args.device)
+
+    def do_run(self, command, **kwargs):
+        build_conf = BuildConfiguration(self.cfg.build_dir)
+        load_offset = build_conf.get("CONFIG_FLASH_LOAD_OFFSET", 0)
+        command = ("ecpprog", "-o", hex(load_offset), self.cfg.bin_file)
+        self.logger.debug(" ".join(command))
+        self.check_call(command)

--- a/scripts/west_commands/tests/test_imports.py
+++ b/scripts/west_commands/tests/test_imports.py
@@ -21,6 +21,7 @@ def test_runner_imports():
         'canopen',
         'dediprog',
         'dfu-util',
+        'ecpprog',
         'esp32',
         'ezflashcli',
         'gd32isp',

